beanie/documents.py

```diff
@@ -87,12 +92,19 @@ async def create(self) -> "Document":
         Create the document in the database
         :return: Document
         """
-        if self.id is not None:
+        if self._is_created:
             raise DocumentAlreadyCreated
+        exclude = None if self.id else {"id"}
         result = await self.get_motor_collection().insert_one(
-            self.dict(by_alias=True, exclude={"id"})
+            self.dict(by_alias=True, exclude=exclude)
         )
-        self.id = PydanticObjectId(result.inserted_id)
+
+        if isinstance(result.inserted_id, ObjectId):
+            self.id = PydanticObjectId(result.inserted_id)
```

Comment:
It looks like it is better to use annotation class to return not only PydanticObjectId, but all of them. Because in another case it will return not the type, which was annotated, and it will confuse. For example, there is the next class:
```python
class TestCase(Document):
    id: UUID = ...
    smth: str
```

In this case, `create()` method will return object with id of type `str` (because json doesn't have UUID types and presents it as a string). 

You can get the id annotation with python inspect module:

```python
class_signature = signature(TestCase)
id_annotation = class_signature.parameters["id"].annotation
```
But the thing is, it can be Optional, or even Union. I think it is too complicated to code own parser here for this case. So you can use the next trick (full working example):

```python
from inspect import signature
from typing import Optional
from uuid import UUID

from pydantic.main import BaseModel


class TestCase(BaseModel): # BaseModel instead of Document for demo, to not init collection
    id: Optional[UUID] = None
    smth: str = "wer"

t = TestCase()

ID_STRING_FROM_DB = "91f02ed2-a07c-4662-8bd1-2f38bc67f0fc"

t.id = ID_STRING_FROM_DB
print(t.id, type(t)) # Here id will be str, even if type is UUID in the annotation

class_signature = signature(TestCase)
id_annotation = class_signature.parameters["id"].annotation


class IdParser(BaseModel):
    id: id_annotation


t.id = IdParser(id=ID_STRING_FROM_DB).id

print(t.id, type(t.id)) # It is UUID now

```

It is a bit tricky and, probably, slow. I don't think it is too slow and will be a bottleneck, but probably here there is room for improvement.